client/components/language-picker/test/modal.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * @format
+ * @jest-environment jsdom
+ */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { shallow } from 'enzyme';
+import { identity, noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { LanguagePickerModal } from '../modal';
+
+describe( 'LanguagePickerModal', () => {
+	const defaultProps = {
+		onSelected: noop,
+		onClose: noop,
+		isVisible: true,
+		languages: [
+			{
+				langSlug: 'en',
+				name: 'English',
+				popular: 1,
+				value: 1,
+				wpLocale: 'en_US',
+			},
+			{
+				langSlug: 'cs',
+				name: 'Čeština',
+				value: 11,
+				wpLocale: 'cs_CZ',
+			},
+			{
+				langSlug: 'it',
+				name: 'Italiano',
+				popular: 8,
+				value: 35,
+				wpLocale: 'it_IT',
+			},
+		],
+		selected: 'en',
+		translate: identity,
+	};
+
+	test( 'should render', () => {
+		const wrapper = shallow( <LanguagePickerModal { ...defaultProps } /> );
+
+		expect( wrapper ).toMatchSnapshot();
+	} );
+
+	describe( 'suggested languages', () => {
+		const browserLanguages = [ 'en-AU', 'en', 'en-US', 'it' ];
```

Comment:
Could we test more combinations here? For example from the code above I think `[ 'en-US', 'it' ]` would only suggest Italian. Let's try even more:
```
[],
[ 'en-US' ],
[ 'en' ],
[ 'it' ],
[ 'it', 'en-US' ],
[ 'it', 'en' ],
[ 'cs', 'it', 'en-US' ]
```

Reply:
We're also testing that we only show supported languages, regardless of the items in browser languages. On that note, I'll expand the mocks so we can extend the tests further. 👍